### PR TITLE
Simplify symbol = symbol predicates in filters

### PR DESCRIPTION
**Pull request by sopel39**
Fixes: https://github.com/prestodb/presto/issues/8806

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestSimplifyExpressions.java [hunk 1]
```diff
@@ -118,6 +118,15 @@ public void testExtractCommonPredicates()
                         " OR (A51 AND A52) OR (A53 AND A54) OR (A55 AND A56) OR (A57 AND A58) OR (A59 AND A60)");
     }
 
+    @Test
+    public void testSimplifiesSymbolEqualsSymbolPredicate()
+    {
+        assertSimplifies("A = A", "A IS NOT NULL");
+        assertSimplifies("(A AND B) = (A AND B)", "(A AND B) IS NOT NULL");
+        assertSimplifies("A = A AND B", "A IS NOT NULL AND B");
+        assertSimplifies("A = A OR B", "A = A OR B");
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestSimplifyExpressions.java]: I think your code does not support things like:
`X OR A = B`

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java [hunk 2]
```diff
@@ -174,6 +180,22 @@ private Expression simplifyExpression(Expression expression)
             ExpressionInterpreter interpreter = ExpressionInterpreter.expressionOptimizer(expression, metadata, session, expressionTypes);
             return LiteralInterpreter.toExpression(interpreter.optimize(NoOpSymbolResolver.INSTANCE), expressionTypes.get(NodeRef.of(expression)));
         }
+
+        private Expression simplifyFilterPredicate(Expression predicate)
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java]: simplify is a wrong name as it does not tell anything. I would name it `replaceSelfEqualityWithNotNull`.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java [hunk 3]
```diff
@@ -140,7 +146,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
                     right,
                     node.getCriteria(),
                     node.getOutputSymbols(),
-                    node.getFilter().map(this::simplifyExpression),
+                    node.getFilter().map(this::simplifyExpression).map(this::simplifyFilterPredicate),
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java]: put your `.map(..)` in separate line. Lets keep a convention of having one `.map` per line.

**Comment by sopel39** [comment]
This rewrite might not be correct

**Pull request closed without merging**